File: frontend/src/types/category.ts
export type TransactionTypeFilter = 'income' | 'expense' | 'both';

export interface Category {
  id: string;
  user_id: string | null;
  name: string;
  icon: string | null;
  color: string | null;
  transaction_type: TransactionTypeFilter;
  is_system: boolean;
  created_at: string;
  updated_at: string;
}

export interface CategoryCreate {
  name: string;
  icon?: string | null;
  color?: string | null;
  transaction_type?: TransactionTypeFilter;
}

export interface CategoryUpdate {
  name?: string;
  icon?: string | null;
  color?: string | null;
  transaction_type?: TransactionTypeFilter;
}

File: backend/app/schemas/category.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class CategoryCreate(BaseModel):
    name: str
    icon: str | None = None
    color: str | None = None
    transaction_type: str = "both"  # income, expense, both


class CategoryUpdate(BaseModel):
    name: str | None = None
    icon: str | None = None
    color: str | None = None
    transaction_type: str | None = None


class CategoryResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID | None
    name: str
    icon: str | None
    color: str | None
    transaction_type: str
    is_system: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
